cmd/gensymbols/walker.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"go/build"
+	"os"
+
+	"golang.org/x/tools/go/packages"
+)
+
+type Walker struct {
+	packages []Pkg
+	context  build.Context
+}
+
+func NewWalker(pkgs []Pkg) Walker {
+	return Walker{
+		packages: pkgs,
+		context:  build.Default,
+	}
+}
+
+func (w Walker) Extract() ([]*packages.Package, error) {
+	var foundPackages []*packages.Package
+	for _, pkg := range w.packages {
+		err := os.Chdir(pkg.Dir)
+		if err != nil {
+			return nil, err
+		}
+
+		dir, err := packages.Load(&packages.Config{
+			Mode: packages.LoadAllSyntax,
```

Comment:
Instead of `LoadAllSyntax`, try `packages.NeedName | packages.NeedImports | packages.NeedTypes`. It should load only what's really, and should speed up the computation quite a bit.

Before, on `x/upgrade/internal/types`:

```
real	0m1,777s
user	0m8,134s
sys	0m2,368s
```

After:

```
real	0m0,698s
user	0m2,229s
sys	0m1,887s
```

Reply:
Good one!!